### PR TITLE
Fail on empty build

### PR DESCRIPTION
**Pull request by grouma**
Closes https://github.com/dart-lang/build/issues/1200

### CODE REVIEW

File: build_runner/lib/src/generate/build_impl.dart [hunk 1]
```diff
@@ -92,7 +92,13 @@ Future<BuildResult> build(
   final buildPhases = await createBuildPhases(
       targetGraph, builders, builderConfigOverrides, isReleaseBuild ?? false);
 
-  var result = await singleBuild(environment, options, buildPhases);
+  BuildResult result;
+  if (buildPhases.isEmpty) {
+    _logger.severe('Nothing was built, yet a build was requested.');
```
Comment (natebosch) [inline build_runner/lib/src/generate/build_impl.dart]: maybe "Nothing can be built, yet a build was requested."

We also want a similar check in watch mode around here: https://github.com/dart-lang/build/blob/0fdde9abb6bacbd8daf02806217278f27dd5b44d/build_runner/lib/src/generate/watch_impl.dart#L92

Comment (grouma) [reply build_runner/lib/src/generate/build_impl.dart]: Added the check. Should we exit as well? My initial thought is no.

File: build_runner/lib/src/generate/watch_impl.dart [hunk 2]
```diff
@@ -92,6 +92,10 @@ Future<ServeHandler> watch(
   final buildPhases = await createBuildPhases(
       targetGraph, builders, builderConfigOverrides, isReleaseBuild ?? false);
 
+  if (buildPhases.isEmpty) {
+    _logger.severe('Nothing can be built, yet a build was requested.');
```
Comment (natebosch) [inline build_runner/lib/src/generate/watch_impl.dart]: can we bail out too?

**Comment by grouma** [comment]
Had to fix some tests. PTAL.

**Pull request merged**